diff --git a/src/g10lib.h b/src/g10lib.h
index a7aee80..51a92f2 100644
--- a/src/g10lib.h
+++ b/src/g10lib.h
@@ -44,7 +44,7 @@
 
 /* Attribute handling macros.  */
 
-#if __GNUC__ > 2 || (__GNUC__ == 2 && __GNUC_MINOR__ >= 5 )
+#if ((__GNUC__ > 2 || (__GNUC__ == 2 && __GNUC_MINOR__ >= 5 )) && ! defined (__MVS__))
 #define JNLIB_GCC_M_FUNCTION 1
 #define JNLIB_GCC_A_NR 	     __attribute__ ((noreturn))
 #define JNLIB_GCC_A_PRINTF( f, a )  __attribute__ ((format (printf,f,a)))
