diff --git a/cipher/Makefile.in b/cipher/Makefile.in
index 521036a..4d88280 100644
--- a/cipher/Makefile.in
+++ b/cipher/Makefile.in
@@ -1410,7 +1410,7 @@ gost-sb.h: gost-s-box
 
 gost-s-box: gost-s-box.c
 	$(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) $(LDFLAGS_FOR_BUILD) \
-	    $(CPPFLAGS_FOR_BUILD) -o $@ $(srcdir)/gost-s-box.c
+	    $(CPPFLAGS_FOR_BUILD) -o $@ $(srcdir)/gost-s-box.c $(LIBS)
 
 # We need to lower the optimization for this module.
 tiger.o: $(srcdir)/tiger.c Makefile
diff --git a/doc/Makefile.in b/doc/Makefile.in
index 356c380..bb8f3e9 100644
--- a/doc/Makefile.in
+++ b/doc/Makefile.in
@@ -916,7 +916,7 @@ uninstall-man: uninstall-man1
 
 yat2m: yat2m.c
 	$(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) $(LDFLAGS_FOR_BUILD) \
-	    $(CPPFLAGS_FOR_BUILD) -o $@ $(srcdir)/yat2m.c
+	    $(CPPFLAGS_FOR_BUILD) -o $@ $(srcdir)/yat2m.c $(LIBS)
 
 .fig.png:
 	fig2dev -L png `test -f '$<' || echo '$(srcdir)/'`$< $@
