diff --git a/mpi/longlong.h b/mpi/longlong.h
index 21bd1a7..2fdfed5 100644
--- a/mpi/longlong.h
+++ b/mpi/longlong.h
@@ -26,7 +26,7 @@ SPDX-License-Identifier: LGPL-2.1-or-later
 #endif
 
 /* On 64-bit, use 128-bit 'unsigned __int128' for UDWtype, if available. */
-#if !defined (UDWtype) && SIZEOF_UNSIGNED___INT128 * 8 == W_TYPE_SIZE * 2
+#if ((!defined (UDWtype)) && ((SIZEOF_UNSIGNED___INT128 * 8) == (W_TYPE_SIZE * 2)) && (!defined(__MVS__)))
 #  define UDWtype unsigned __int128
 #endif
 
@@ -1503,7 +1503,7 @@ extern USItype __udiv_qrnnd ();
 /***************************************
  *********** s390x/zSeries  ************
  ***************************************/
-#if defined (__s390x__) && W_TYPE_SIZE == 64 && __GNUC__ >= 4
+#if defined (__s390x__) && W_TYPE_SIZE == 64 && __GNUC__ >= 4 && !defined(__MVS__)
 # define add_ssaaaa(sh, sl, ah, al, bh, bl) \
   __asm__ ("algr %1,%5\n"                                               \
 	   "alcgr %0,%3\n"                                              \
