diff --git a/random/rndgetentropy.c b/random/rndgetentropy.c
index a6f3c4a..941e0f6 100644
--- a/random/rndgetentropy.c
+++ b/random/rndgetentropy.c
@@ -96,7 +96,11 @@ _gcry_rndgetentropy_gather_random (void (*add)(const void*, size_t,
                * RHEL kernel.
                */
               nbytes = length < 32 ? length : 32;
-              ret = getrandom (buffer, nbytes, GRND_RANDOM);
+	      #ifdef __MVS__
+	      	ret = getentropy (buffer, nbytes);
+	      #else
+              	ret = getrandom (buffer, nbytes, GRND_RANDOM);
+	      #endif
             }
           else
 #endif
